### DISCUSSION TITLE
Frigate 0.14 beta2 "image/webp not valid"

### DISCUSSION THREAD

**Discussion by arpel** [skrashevich/double-take#519] [Q&A]
Hi !
As it may soon be released as stable I started to play with Frigate 0.14 : since then I had the feeling that I was getting much less "snapshots" (images in the "Matches" tab) and I just saw that almost for each event I have the following error in the double-take log (last package) : 

```
24-06-01 19:12:52 info: processing hik_1: 1717269167.086175-gischx
24-06-01 19:12:52 error: url validation failed for latest: http://192.168.0.61:5000/api/hik_1/latest.jpg?h=720 - image/webp not valid
24-06-01 19:12:55 info: done processing hik_1: 1717269167.086175-gischx in 2.97 sec
```

Do yo have any hints ?

**Comment by pdobrien3** [comment]
I have a similar issue although I am not seeing errors in the logs.  Double-take is not processing latest.jpg images from frigate on the 0.14 beta

**Comment by arpel** [comment]
I think I've found the issue and a workaround ; Frigate 0.14+ seems to be using webp instead of png/jpg for some of the snapshot (not clear if that general or not yet).

I've come up with a "working" solution (tested on my install) : https://github.com/skrashevich/double-take/pull/534

**Reply by jsapede** [reply]
tried to implement your changes but got following errors : 

`24-07-05 08:28:06 info: processing Portail: 1720160885.694322-2nz88p
24-07-05 08:28:10 error: stream error: Request failed with status code 404
24-07-05 08:28:10 error: An error occurred while processing promises: Request failed with status code 404 {
  message: 'Request failed with status code 404',
  name: 'AxiosError',
  stack: 'AxiosError: Request failed with status code 404\n' +
    '    at settle (/double-take/api/node_modules/axios/dist/node/axios.cjs:1967:12)\n' +
    '    at IncomingMessage.handleStreamEnd (/double-take/api/node_modules/axios/dist/node/axios.cjs:3066:11)\n' +
    '    at IncomingMessage.emit (node:events:529:35)\n' +
    '    at endReadableNT (node:internal/streams/readable:1400:12)\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)\n' +
    '    at Axios.request (/double-take/api/node_modules/axios/dist/node/axios.cjs:3877:41)\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
    '    at async module.exports.stream (/double-take/api/src/util/process.util.js:243:24)\n' +
    '    at async module.exports.polling (/double-take/api/src/util/process.util.js:34:22)\n' +
    '    at async Promise.all (index 1)\n' +
    '    at async module.exports.start (/double-take/api/src/controllers/recognize.controller.js:155:23)',
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [ 'xhr', 'http' ],
    transformRequest: [ null ],
    transformResponse: [ null ],
    timeout: 5000,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    env: {},
    headers: {
      Accept: 'application/json, text/plain, */*',
      'User-Agent': 'axios/1.6.7',
      'Accept-Encoding': 'gzip, compress, deflate, br'
    },
    method: 'get',
    url: 'http://192.168.0.112:5000/api/events/1720160885.694322-2nz88p/snapshot.jpg?h=500',
    responseType: 'arraybuffer'
  },
  code: 'ERR_BAD_REQUEST',
  status: 404
}
24-07-05 08:28:10 info: done processing Portail: 1720160885.694322-2nz88p in 3.44 sec
24-07-05 08:28:10 error: An error occurred at TypeError: Cannot read properties of undefined (reading 'forEach')
    at /double-take/api/src/controllers/recognize.controller.js:210:26
    at Array.forEach (<anonymous>)
    at module.exports.start (/double-take/api/src/controllers/recognize.controller.js:209:39)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
24-07-05 08:28:10 error: An error occurred when recogniting file: Cannot read properties of undefined (reading 'forEach')`

**Reply by arpel** [reply]
Hi @jsapede, I'm sorry but my change (based on latest release 1.13.11.8) did not touch the mentioned files (axios, recognize ..).
The only one modified and listed in your error is process.utils.js but not at all at those locations (and the modification to this file shall not introduce any error : that's only a additional value in an array).

**Reply by jsapede** [reply]
after having a deeper look in sources i found a difference in your source : 

`
  const request = await axios({
    method: 'get',
    url: `http://0.0.0.0:${SERVER.PORT}${UI.PATH}/api/storage/matches/${originalFilename}?box=true`,
    headers: AUTH ? { authorization: jwt.sign({ route: 'storage' }) } : null,`


and on the repo source for rc3 : 

`  const request = await axios({
    method: 'get',
    url: `http://${SERVER.HOST}:${SERVER.PORT}${UI.PATH}/api/storage/matches/${originalFilename}?box=true`,
    headers: AUTH ? { authorization: jwt.sign({ route: 'storage' }) } : null,`


IP is set to 0.0.0.0 in your file. would this be the cause ?

**Reply by arpel** [reply]
Sorry but this has nothing to do with my changes which only modify the image handling process and not any networking stuff.
I'll suggest to _backport_ my edits manually into your working code base as I'm pretty sure this is linked to other changes.

**Reply by jsapede** [reply]
uderstood it isnt in link with your modifications but the IP change int he rc3 was the source of the problem i encountered.
switching back to 0.0.0.0 instead of ${SERVER.HOST} did the trick and now your modifications work !
for the ones who could encounter the same situation

**Comment by dz4news** [comment]
Any idea when a new version of double take will be available to address this issue?
